### ISSUE TITLE
Conversation tatams appear twice

### ISSUE THREAD

**Issue by rlheritier** [ippontech/tatami#859]
In this screenshot, Justin's tatam appears twice. 

![screen shot 2015-04-17 at 10 13 56 am](https://cloud.githubusercontent.com/assets/4593279/7203730/b66e032c-e4ea-11e4-9fe3-b24d76a2e3e6.png)

**Comment by JustinRisch** [comment]
That is working as intended. (it will only happen when a reply receives a reply)

**Comment by jxerome** [comment]
*I move my comment from #853 because It is more appropriate here.*


I think the idea is to present individual messages (this is a twitter, not a facebook clone) but add some context that allow you to understand messages without have to deploy the whole conversation.

I like the idea.

However, the implementation is a bit confusing because the real message is indented and the context looks to be the main message. Except for first message in the conversation which looks different.

I think that a simple change in the presentation should stop the confusion.

**Comment by mleneveut** [comment]
I think this is hard to read, and conversations are so very hard to follow.

Would it be technicaly possible to add all replies under the original post ? I understand it could be complicate to know what is the (great great great grand)parent tweet, but it would be better IMO.

**Comment by JustinRisch** [comment]
If you want to read it from top to bottom, you can click the "view conversation" link, which will show a linear conversation. 
Supposing 1, 2, 3, and 4 are posts. 2 and 3 reply to 1. 4 replies to 2. 

Regular view: 
2
- 4
1
- 2
- 3

Conversation view (when clicking 4)
1
2
4

**Comment by rlheritier** [comment]
I would like to see all the replies under the original post too.

**Comment by jxerome** [comment]
Well having all answers in the same block would switch tatami from message orientation (twitter like) to conversation orientation (facebook like). Having the same conversation repeated ever and ever with only few more replies would make last replies lost in a ocean of old data.

I think we should keep the message orientation with just a citation of the replied tatam to give the context. Going to the full conversation screen is a good way to get the rest of the conversation if you want more.

**Comment by JustinRisch** [comment]
It's certainly up for changing. Can we get a mock up of what you're looking for though, not sure I fully understand what third option there is. (full conversation vs. message and it's children)

**Comment by JustinRisch** [comment]
So I have an idea that may work... 
![mockup](https://cloud.githubusercontent.com/assets/6710878/7208460/fc0595fc-e50e-11e4-8197-9ddeff20a01f.png)
In the above example, I'm displaying the initial post (white). When someone comments, it's in blue-grey. 

All comments to the original are shown, directly under the initial post. Not indented, doesn't change much else. 

Secondary posts get trickier. Rather than showing *all* sub-comments, I've decided to show them like this: 

![cherryblossomui](https://cloud.githubusercontent.com/assets/6710878/7208499/4e691c92-e50f-11e4-814a-32815d31382c.png)

Red boxes are shown, grey ones in the "view full conversation" mode. We will change conversation mode so that it stops at the node you clicked "view conversation" on, and treats that as a new "initial post" for display purposes (temporarily). (or from a computer science background: shows a list of conversations leading to it, then it as a root node, and displays the first level children and all leaves, as in the above diagram. 

This design allows us to do a couple things: 
1. All replies are kept under the initial status, so no more duplicates. 
2. It remains about the "messages", as more current replies move older ones from display.
3. gives a unique feel, as you're able to navigate through conversation, following branches on the tree, with each branch representing it's own conversation. 
4. It shows the 3 most important things about a conversation-- what sparked it, who engaged that conversation, and what conclusion they came to (if any). 

Better?

**Comment by jxerome** [comment]
Why not something simplier as this ?
It's not very different from first screenshot proposal but easier to understand.
<img width="714" alt="tatam-flow-1" src="https://cloud.githubusercontent.com/assets/1355634/11058214/795a3a38-8791-11e5-922f-147dfbc7c27f.png">

**Comment by jxerome** [comment]
Because the reply to reference is more a context reminder that should not hide the new tatam, we could truncate very long ones. 5 lines seems good enough.

For more context, the user shall click on the "View Conversation" link.

**Comment by rlheritier** [comment]
I liked how it worked when we went live with the AngularJS version. It was simple. We broke it because it introduced another side-effect, but I liked it. It was inspired by Twitter. Twitter had to do the same thing in 2012/2013.

**Comment by FatenHabachi** [comment]
I like the Justin's proposition. 
It's visual + hierarchical = easy to read !

**Comment by rlheritier** [comment]
We need a way to quickly prototype it anyway. These things are hard to imagine without a real conversation in the tool, in a sandbox.

**Comment by jxerome** [comment]
@FatenHabachi The hierarchical view is the purpose of the "View conversation" view not of the timeline. The timeline should show only the new tatam with enough context to understand where it comes from.

**Comment by jxerome** [comment]
Nowadays, I would say the best UI would be inspired from IRC: messages in chronological order grouped in thematic channels. Slack uses this model with a great success.

**Comment by yhippa** [comment]
![capture](https://cloud.githubusercontent.com/assets/420309/11633021/66d9d066-9cd7-11e5-99b8-0f12503419bd.PNG)

On the current version of Twitter I initially replied to the initial "Test" tweet with "Test 1", replied to myself again with "Test 2", replied to "Test 2" with "Test 2-1", and then finally replied to the original tweet with "Test 3".  As you can see it is basically displaying it all in chronological order and in a flat fashion.

I am leaning towards Justin's approach mostly because I (and apparently other people) have a hard time following conversations but am also concerned about ridiculous nesting.

**Comment by rlheritier** [comment]
Regarding ‘ridiculous nesting’, we can just limit the depth like Facebook does.
I think you can only reply to replies, on Facebook. You can’t go deeper.

Romain
